src/main/java/com/squadb/workassistantapi/reservation/application/DefaultReservationValidator.java

```diff
@@ -0,0 +1,66 @@
+package com.squadb.workassistantapi.reservation.application;
+
+import com.squadb.workassistantapi.book.domain.Book;
+import com.squadb.workassistantapi.member.domain.Member;
+import com.squadb.workassistantapi.reservation.domain.*;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+@Component
+@RequiredArgsConstructor
+class DefaultReservationValidator implements ReservationValidator {
+
+    private final ReservationRepository reservationRepository;
+
+    @Override
+    public void canReserve(Member member, Book book) {
+        List<Reservation> reservationList = findAllWaitingReservationByBookId(book.getId());
+        validateNotExistsMyReservation(member, reservationList);
+        validateNotOverThanMaxReservationCountPerBook(reservationList);
+        validateNotOverThanMaxReservationCountPerMember(member.getId());
+    }
+
+    private List<Reservation> findAllWaitingReservationByBookId(Long bookId) {
+        return reservationRepository.findAllByBookIdAndStatus(bookId, ReservationStatus.WAITING);
+    }
+
+    private void validateNotExistsMyReservation(Member member, List<Reservation> reservationList) {
```

Comment:
```suggestion
    private void validateNotExistsMyReservation(Member member, List<Reservation> reservations) {
```
제가 넥스트스텝 강의를 진행하면서 배운 것이 있는데, 자료구조에 맞게 변수명을 짓는 것은 좋지 않다고 해요.
자료구조는 언제 어떻게 바뀔지 모르기 때문에 Collection에 해당하는 변수명은 reservation**s**와 같이 짓는게 좋다고 합니다.
제 기억으로 미리디 사내에서 네이밍 컨벤션이 ~List로 될 때가 종종 있었던 거 같은데, 토이 프로젝트에서는 s를 붙여보는 게 좋은 거 같아요!

Reply:
저도 미리디 위키 문서에서 ~s 대신 ~list 로  쓰자는 글을 본것 같아서 list로 사용했는데
Collection이 바뀔 때 변수명이 어울리지 않을 것 같기도하네요. ~s 로 수정하겠습니다.